In [10]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [11]:
dataset = pd.read_csv('50_Startups.csv')
x = dataset.iloc[:,:-1].values
y= dataset.iloc[:,-1].values
# print(x)

In [12]:
# changing the categorical data into numeric values
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder
ct = ColumnTransformer(transformers=[('encoder',OneHotEncoder(),[3])] , remainder='passthrough')
x = np.array(ct.fit_transform(x))
# print(x)

In [18]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train,y_test = train_test_split(x,y,test_size=0.2,random_state =0)


In [19]:
# training model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [20]:
# comparison between predicted and real values
y_pred = regressor.predict(x_test)
np.set_printoptions(precision=2) #to avoid long decimal numbers
print(np.concatenate((y_pred.reshape(len(y_pred),1),y_test.reshape(len(y_test),1)),1))

[[103015.2  103282.38]
 [132582.28 144259.4 ]
 [132447.74 146121.95]
 [ 71976.1   77798.83]
 [178537.48 191050.39]
 [116161.24 105008.31]
 [ 67851.69  81229.06]
 [ 98791.73  97483.56]
 [113969.44 110352.25]
 [167921.07 166187.94]]


In [22]:
# prediction
# predict expects a 2D array
value = regressor.predict([[1,0,0,15000,30000,20000]])
print(value)

[55874.91]


In [25]:
# to get the equation
print(regressor.intercept_)
print(regressor.coef_)

42467.52924853204
[ 8.66e+01 -8.73e+02  7.86e+02  7.73e-01  3.29e-02  3.66e-02]


So,

Profit=86.6×Dummy State1 −873×Dummy State 2 +786×Dummy State 3  −0.773×R&D Spend  +0.0329×Administration  +0.0366×Marketing Spend  +42467.53